kernels/test/op_grid_sampler_2d_test.cpp

```diff
@@ -0,0 +1,374 @@
+//
+//  Copyright (c) 2025 Apple Inc. All rights reserved.
+//  Provided subject to the LICENSE file in the top level directory.
+//
+
+#include <executorch/kernels/test/FunctionHeaderWrapper.h> // Declares the operator
+#include <executorch/kernels/test/TestUtil.h>
+#include <executorch/kernels/test/supported_features.h>
+#include <executorch/runtime/core/exec_aten/exec_aten.h>
+#include <executorch/runtime/core/exec_aten/testing_util/tensor_factory.h>
+#include <executorch/runtime/core/exec_aten/testing_util/tensor_util.h>
+#include <gtest/gtest.h>
+
+using executorch::aten::ScalarType;
+using executorch::aten::Tensor;
+using torch::executor::testing::TensorFactory;
+
+class OpGridSampler2dTest : public OperatorTest {
+ protected:
+  // Test grid_sampler_2d with bilinear interpolation
+  void test_grid_sampler_2d_bilinear() {
+    TensorFactory<ScalarType::Float> tf;
+
+    // Create a 1x1x2x2 input tensor
+    const std::vector<int32_t> input_sizes = {1, 1, 2, 2};
+    std::vector<float> input_data = {1.0, 2.0, 3.0, 4.0};
+    Tensor input = tf.make(input_sizes, input_data);
```

Comment:
nit: you don't need to name input_sizes and input_data here, just inline the initializer lists